DopplerCurrencyJob/DopplerCurrencyJob.cs

```diff
@@ -25,27 +29,98 @@ public class DopplerCurrencyJob
             _dopplerSapService = dopplerSapService;
         }
 
+
         [AutomaticRetry(OnAttemptsExceeded = AttemptsExceededAction.Delete, Attempts = 0)]
-        public IList<CurrencyResponse> Run() => RunAsync().GetAwaiter().GetResult();
+        public DopplerCurrencyJobResponse Run() => RunAsync().GetAwaiter().GetResult();
 
-        private async Task<IList<CurrencyResponse>> RunAsync()
+        private async Task<DopplerCurrencyJobResponse> RunAsync()
         {
-            _logger.LogInformation("Getting currency per each code enabled.");
-            var currencyDto = await _dopplerCurrencyService.GetCurrencyByCode();
+            var taskFailed = false;
+
+            var resultBuffer = new StringBuilder();
+            var message = "Getting currency per each code enabled.";
+            _logger.LogInformation(message);
+            resultBuffer.Append(message).AppendLine();
```

Comment:
What do you think about moving all of this into a method to avoid the auxiliary variable `message`? 

```suggestion
            LogAndAppendToStringBuilder(resultBuffer, "Getting currency per each code enabled.");
```

Reply:
it is a good idea!

Reply:
Hecho, marco como resuelto.